packages/obonode/obojobo-sections-assessment/server/attempt-review.js

```diff
@@ -3,48 +3,51 @@ const DraftModel = require('obojobo-express/server/models/draft')
 const { getFullQuestionsFromDraftTree } = require('./util')
 const logger = require('obojobo-express/server/logger')
 
-const getQuestionModelsFromAttempt = async attemptId => {
-	const attempt = await Assessment.getAttempt(attemptId)
-
-	// @TODO: memoize or cache this
-	const draftDocument = await DraftModel.fetchDraftByVersion(
-		attempt.draft_id,
-		attempt.draft_content_id
-	)
-
-	const attemptQuestionModels = getFullQuestionsFromDraftTree(draftDocument, attempt.state.chosen)
-	const attemptQuestionModelsMap = {}
-	for (const questionModel of attemptQuestionModels) {
-		attemptQuestionModelsMap[questionModel.id] = questionModel
-	}
+const getQuestionModelsFromAttempt = async (attemptId, userId) => {
```

Comment:
This changed a lot, now requires user id and errors are caught internally (so that if one attempt request fails, the others will at least show up)